From 405fccdf09f8fb69024391cbe41d529a5b10e472 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:02:14 -0400
Subject: [usb] xhci: correct the usage of USB_CTRL_SET_TIMEOUT

Message-id: <1410887463-20674-63-git-send-email-dzickus@redhat.com>
Patchwork-id: 95203
O-Subject: [RHEL7 PATCH 062/591] xhci: correct the usage of USB_CTRL_SET_TIMEOUT
Bugzilla: 1032371
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1032371

commit d194c031994d3fc1038fa09e9e92d9be24a21921
Author: xiao jin <jin.xiao@intel.com>
Date:   Fri Oct 11 08:57:03 2013 +0800

    xhci: correct the usage of USB_CTRL_SET_TIMEOUT

    The usage of USB_CTRL_SET_TIMEOUT in xhci is incorrect.
    The definition of USB_CTRL_SET_TIMEOUT is 5000ms. The
    input timeout to wait_for_completion_interruptible_timeout
    is jiffies. That makes the timeout be longer than what
    we want, such as 50s in some platform.

    The patch is to use XHCI_CMD_DEFAULT_TIMEOUT instead of
    USB_CTRL_SET_TIMEOUT as command completion event timeout.

    Signed-off-by: xiao jin <jin.xiao@intel.com>
    Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/xhci-hub.c b/drivers/usb/host/xhci-hub.c
index 45ca532..805f234 100644
--- a/drivers/usb/host/xhci-hub.c
+++ b/drivers/usb/host/xhci-hub.c
@@ -296,7 +296,7 @@ static int xhci_stop_device(struct xhci_hcd *xhci, int slot_id, int suspend)
  /* Wait for last stop endpoint command to finish */
  timeleft = wait_for_completion_interruptible_timeout(
    cmd->completion,
-   USB_CTRL_SET_TIMEOUT);
+   XHCI_CMD_DEFAULT_TIMEOUT);
  if (timeleft <= 0) {
   xhci_warn(xhci, "%s while waiting for stop endpoint command\n",
     timeleft == 0 ? "Timeout" : "Signal");
diff --git a/drivers/usb/host/xhci.c b/drivers/usb/host/xhci.c
index ab5de14..9ecd82a 100644
--- a/drivers/usb/host/xhci.c
+++ b/drivers/usb/host/xhci.c
@@ -3459,7 +3459,7 @@ int xhci_discover_or_reset_device(struct usb_hcd *hcd, struct usb_device *udev)
  /* Wait for the Reset Device command to finish */
  timeleft = wait_for_completion_interruptible_timeout(
    reset_device_cmd->completion,
-   USB_CTRL_SET_TIMEOUT);
+   XHCI_CMD_DEFAULT_TIMEOUT);
  if (timeleft <= 0) {
   xhci_warn(xhci, "%s while waiting for reset device command\n",
     timeleft == 0 ? "Timeout" : "Signal");
-- 
1.7.1